File: JVM/jnetcall-java-tests/src/test/java/jnetcall/java/tests/io/MemTransportTest.java
package jnetcall.java.tests.io;

import org.javatuples.Pair;

import jnetbase.java.io.MemPipeStream;
import jnetcall.java.api.io.ISendTransport;
import jnetcall.java.impl.io.StreamTransport;

public final class MemTransportTest extends TransportTest {

    @Override
    protected Pair<ISendTransport, ISendTransport> getBoth() {
        MemPipeStream mem11001 = new MemPipeStream();
        MemPipeStream mem11002 = new MemPipeStream();
        StreamTransport left = new StreamTransport(
                Encoding,
                mem11001.asI(),
                mem11002.asO()
        );
        StreamTransport right = new StreamTransport(
                Encoding,
                mem11002.asI(),
                mem11001.asO()
        );
        return Pair.with(left, right);
    }
}

File: NET/JNetCall.Sharp.Tests/IO/MemTransportTest.cs
using System.IO;
using JNetBase.Sharp.IO;
using JNetCall.Sharp.API.IO;
using JNetCall.Sharp.Impl.IO;

namespace JNetCall.Sharp.Tests.IO
{
    public sealed class MemTransportTest : TransportTest
    {
        protected override (ISendTransport, ISendTransport) GetBoth()
        {
            Stream mem11001 = new MemPipeStream();
            Stream mem11002 = new MemPipeStream();
            var left = new StreamTransport(
                Encoding,
                mem11001,
                mem11002
            );
            var right = new StreamTransport(
                Encoding,
                mem11002,
                mem11001
            );
            return (left, right);
        }
    }
}